reporting.md

```diff
@@ -0,0 +1,615 @@
+    tabulating correctly.
+
+  
+### Routing Failures
+  * `mx-mismatch`: This indicates that the MX resolved for the recipient domain
+    did not match the MX constraint specified in the policy.
+  * `certificate-mismatch`: This indicates that the certificate presented by the
+    receiving MX did not match the MX hostname
+
+### Negotiation Failures
+
+  * `starttls-not-supported`: This indicates that the recipient MX did not
+    support STARTTLS.
+  * `invalid-certificate`: This indicates that the certificate presented by the
+    receiving MX did not validate according to the policy validation constraint.
+    (Either it was not signed by a trusted CA or did not match the DANE TLSA
+    record for the recipient MX.)
+  * `expired-certificate`: This indicates that the certificate has expired.
+
+#### DANE-specific Policy Failures
+  * `tlsa-invalid`: This indicates a validation error for Policy Domain
+    specifying "tlsa" validation.
+  * `dnssec-invalid`: This indicates a failure to validate DNS records for a
+    Policy Domain with a published "tlsa" record.
+
+#### STS-specific Policy Failures
+  * `sts-invalid`: This indicates a validation error for Policy Domain
+    specifying "STS" validation.
+
+  * sender-does-not-support-validation-method: This indicates the sending system
+    can never validate using the requested validation mechanism.
+
+
+# Reporting Delegation
+
+   It is possible to specify destinations for the different reports that
+   are outside the authority of the Policy Domain making the request.
+   This allows domains that do not operate mail servers to request
+   reports and have them go someplace that is able to receive and
+   process them.
+
+   Without checks, this would allow a bad actor to publish a TLS-RPT policy
+   record that requests that reports be sent to a victim address, potentially
+   causing the victim to be flooded with unwanted reports. Therefore, a
+   verification mechanism is included as inspired by Section 7.1 from 
+   [@!RFC7489].
+
+   When a sending MTA discovers an TLS-RPT policy in the DNS, and the
+   Organizational Domain at which that record was discovered is not
+   identical to the Organizational Domain of the host part of the
+   authority component of a [URI] specified in the "rua" or "ruf" tag,
+   the following verification steps are to be taken:
+
+   1.  Extract the host portion of the authority component of the URI.
+       Call this the "destination host", as it refers to a Report
+       Receiver.
+
+   2.  Prepend the string "_report._tlsrpt".
+
+   3.  Prepend the domain name from which the policy was retrieved,
+       after conversion to an A-label if needed.
+
+   4.  Query the DNS for a TXT record at the constructed name.  If the
+       result of this request is a temporary DNS error of some kind
+       (e.g., a timeout), the Mail Receiver MAY elect to temporarily
+       fail the delivery so the verification test can be repeated later.
```

Comment:
fixed. But this section is still not done, because I got halfway through and then started wondering whether it really matters. That I would publish a bogus STS policy as a way to get senders to reflect bogus reports to a victim seems a little far-fetched...